ydb/docs/en/core/integrations/orm/jooq.md

```diff
@@ -0,0 +1,187 @@
+                <excludes>.sys.*</excludes>
+            </database>
+            <target>
+                <packageName>ydb</packageName>
+                <directory>./src/main/java</directory>
+            </target>
+        </generator>
+    </configuration>
+</plugin>
+```
+
+Example of generated classes from [YQL tutorial](../../dev/yql-tutorial/create_demo_tables.md):
+
+```
+ydb/DefaultCatalog.java
+ydb/default_schema
+ydb/default_schema/tables
+ydb/default_schema/tables/Seasons.java
+ydb/default_schema/tables/records
+ydb/default_schema/tables/records/SeriesRecord.java
+ydb/default_schema/tables/records/EpisodesRecord.java
+ydb/default_schema/tables/records/SeasonsRecord.java
+ydb/default_schema/tables/Series.java
+ydb/default_schema/tables/Episodes.java
+ydb/default_schema/Indexes.java
+ydb/default_schema/Keys.java
+ydb/default_schema/Tables.java
+ydb/default_schema/DefaultSchema.java
+```
+
+## Usage {#usage}
+
+To integrate {{ ydb-short-name }} with JOOQ into your project, you need to add two dependencies: {{ ydb-short-name }} JDBC Driver and the JOOQ extension for {{ ydb-short-name }}.
+
+Examples for different build systems:
+
+{% list tabs %}
+
+- Maven
+
+    ```xml
+    <!-- Set actual versions -->
+    <dependency>
+        <groupId>tech.ydb.jdbc</groupId>
+        <artifactId>ydb-jdbc-driver</artifactId>
+        <version>${ydb.jdbc.version}</version>
+    </dependency>
+
+    <dependency>
+        <groupId>tech.ydb.dialects</groupId>
+        <artifactId>spring-data-jdbc-ydb</artifactId>
+        <version>${spring.data.jdbc.ydb}</version> 
+    </dependency>
+    ```
+
+- Gradle
+
+    ```groovy
+    dependencies {
+        // Set actual versions
+        implementation "tech.ydb.dialects:spring-data-jdbc-ydb:$ydbDialectVersion"
+        implementation "tech.ydb.jdbc:ydb-jdbc-driver:$ydbJdbcVersion"
+    }
+    ```
+
+{% endlist %}
+
+To obtain a `YdbDSLContext` (an extension of `org.jooq.DSLContext`), use the `tech.ydb.jooq.YDB` class. For example:
+
+```java
+String url = "jdbc:ydb:<grpc/grpcs>://<host>:<2135/2136>/path/to/database[?saFile=file:~/sa_key.json]";
+Connection conn = DriverManager.getConnection(url);
+
+YdbDSLContext dsl = YDB.using(conn);
+```
+
+or
+
+```java
+String url = "jdbc:ydb:<grpc/grpcs>://<host>:<2135/2136>/path/to/database[?saFile=file:~/sa_key.json]";
+try(CloseableYdbDSLContext dsl = YDB.using(url)) {
+        // ...
+}
+```
+
+Extend `JooqAutoConfiguration.DslContextConfiguration` with your own `YdbDSLContext`. For example:
+
+```java
+@Configuration
+public class YdbJooqConfiguration extends JooqAutoConfiguration.DslContextConfiguration {
+
+    @Override
+    public YdbDSLContextImpl dslContext(org.jooq.Configuration configuration) {
+        return YdbDSLContextImpl(configuration);
+    }
+}
+```
+
+```properties
+spring.datasource.driver-class-name=tech.ydb.jdbc.YdbDriver
+spring.datasource.url=jdbc:ydb:<grpc/grpcs>://<host>:<2135/2136>/path/to/database[?saFile=file:~/sa_key.json]
```

Comment:
If this indeed is intended to be used inside the Spring Framework, let's make separate examples for each scenario and explicitly explain what options are available.